### DISCUSSION TITLE
HIde/Show code in HTML Ouput

### DISCUSSION THREAD

**Discussion by ronbactawar** [fonsp/Pluto.jl#1233] [Pluto development] [answered]
**Problem:**  Prior to exporting a Pluto notebook to HTML, the coder must decide what code to hide or show.   As an example, giving a business executive an HTML document filled with potentially distracting code or conversely a code literate reviewer a document without much code is not a really unified approach.  I feel that it is more practical to let the user choose what code (if any) they wish to observe in the output HTML.  

**Possible Solution:**  An output HTML document that initially shows no code but is equipped with a button that can toggle show-all or hide-all code; or multiple buttons at each code section that that can hide/expand code.  As an example, R markdown has an implementation that fixes this problem.  

R markdown has code block folding which is documented here:
https://bookdown.org/yihui/rmarkdown-cookbook/fold-show.html

A discussion of the importance of code block folding in R markdown:
https://www.youtube.com/watch?v=w_97Vr-d3yM

**Comment by fonsp** [comment, marked as answer]
Try this:
```julia
showhidebutton = html"""
<style>
body.hide-all-code pluto-input {
	display: none !important;
}
</style>

<button onClick="document.body.classList.toggle('hide-all-code')">Show/hide all code</button> 
"""
```

**Reply by ronbactawar** [reply]
Thanks you for your answer fonsp; that HTML chunk worked wonderfully well!  On a minor note it possible to set the default state of the button to hidden, so that no code is shown initially? If that’s not possible no worries.

**Reply by fonsp** [reply]
Add this:
```

<script>
document.body.classList.add('hide-all-code')
</script>
```

**Reply by ronbactawar** [reply]
Much appreciated, that did the job!
